apps/passport-server/test/devconnect.spec.ts

```diff
@@ -373,6 +381,218 @@ describe("devconnect functionality", function () {
+    const devconnectPretixAPIConfigFromDB = await getDevconnectPretixConfig(db);
+    if (!devconnectPretixAPIConfigFromDB) {
+      throw new Error("Could not load API configuration");
+    }
+
+    const organizer = devconnectPretixAPIConfigFromDB?.organizers[0];
+    const orgUrl = organizer.orgURL;
+
+    // Pick an event where we will consume all of the tickets
+    const eventID = organizer.events[0].eventID;
+    const eventConfigID = organizer.events[0].id;
+    const org = mocker.get().organizersByOrgUrl.get(orgUrl) as IOrganizer;
+
+    // Simulate Pretix returning tickets as being checked in
+    server.use(
+      rest.get(orgUrl + `/events/:event/orders`, (req, res, ctx) => {
+        const skip = (req.params.event as string) !== eventID;
+        const orders = skip
+          ? org.ordersByEventID.get(eventID)
+          : org.ordersByEventID.get(eventID)?.map((order) => {
+              return {
+                ...order,
+                positions: order.positions.map((position) => {
+                  return {
+                    ...position,
+                    checkins: [{ type: "entry", datetime: new Date() }]
+                  };
+                })
+              };
+            });
+
+        return res(
+          ctx.json({
+            results: orders,
+            next: null
+          })
+        );
+      })
+    );
+
+    // Set up a sync manager for a single organizer
+    const os = new OrganizerSync(
+      organizer,
+      new DevconnectPretixAPI({ requestsPerInterval: 300 }),
+      application.services.rollbarService,
+      application.context.dbPool
+    );
+
+    expect(await os.run()).to.not.throw;
+
+    const tickets = await fetchDevconnectPretixTicketsByEvent(
+      db,
+      eventConfigID
+    );
+
+    // All tickets for the event should be consumed
+    expect(tickets.length).to.eq(
+      tickets.filter(
+        (ticket: DevconnectPretixTicket) => ticket.is_consumed === true
+      ).length
+    );
+  });
+
+  /**
+   * This covers the case where we have a ticket marked as consumed, but
+   * the check-in is cancelled in Pretix.
+   */
+  step("should be able to un-check-in a ticket via sync", async function () {
+    const devconnectPretixAPIConfigFromDB = await getDevconnectPretixConfig(db);
+    if (!devconnectPretixAPIConfigFromDB) {
+      throw new Error("Could not load API configuration");
+    }
+
+    const organizer = devconnectPretixAPIConfigFromDB?.organizers[0];
+
+    const eventConfigID = organizer.events[0].id;
+
+    // Set up a sync manager for a single organizer
+    const os = new OrganizerSync(
+      organizer,
+      new DevconnectPretixAPI({ requestsPerInterval: 300 }),
+      application.services.rollbarService,
+      application.context.dbPool
+    );
+
+    // Because we're not patching the data from Pretix, default responses
+    // have no check-ins
+    expect(await os.run()).to.not.throw;
```

Comment:
worth separating the comment out from the line below it and adding a sentence more of description